app/src/main/java/org/schabi/newpipe/util/Localization.java

```diff
@@ -328,7 +328,7 @@ public static String relativeTime(final Calendar calendarTime) {
     // This method is a modified version of GregorianCalendar.from(ZonedDateTime).
     // Math.addExact() and Math.multiplyExact() are desugared even though lint displays a warning.
     @SuppressWarnings("NewApi")
-    private static GregorianCalendar fromUTC(final OffsetDateTime offsetDateTime) {
+    public static GregorianCalendar fromUTC(final OffsetDateTime offsetDateTime) {
```

Comment:
Don't make a private method public for the sake of testing.

Instead you should test the methods thats using the private method or extract the method into it's own class (not 100% sure on the second part myself yet)

In this case i would write tests for `public static String relativeTime(final OffsetDateTime offsetDateTime)`. If done this way you will provide the value of having regression test for when the workaround is removed. Because currently your tests will need to be removed, when the workaround ist removed.

Reply:
I tried creating an instrumented test, but attempting to run it results in an error message saying that the test class cannot be found.

Reply:
I'll open an pr to your branch tonight/tomorrow with what i mean. Maybe i missed something, that makes what i mean not possible.

Reply:
No, I get that you want to test `Localization.relativeTime()` directly, but I can't run the instrumented test, so I tested PrettyTime manually.

Reply:
Ok, now i understand why you need an instrumented test. The dependency are direcly accessed and not injected. This would need some major refactoring, which can be done in another PR.

Reply:
I created a PR with Dagger, so I could add a commit for DI with `Localization` there.